sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java

```diff
@@ -16,37 +17,107 @@
+            return ContentType.IMAGE_TIFF;
+        } else {
+            throw new IllegalArgumentException(
+                "Content type could not be detected. Should use other overload API that takes content type.");
+        }
+    }
+
+    /**
+     * Creates a Flux of ByteBuffer, with each ByteBuffer wrapping bytes read from the given
+     * InputStream.
+     *
+     * @param inputStream InputStream to back the Flux
+     * @return Flux of ByteBuffer backed by the InputStream
+     */
+    public static Flux<ByteBuffer> toFluxByteBuffer(InputStream inputStream) {
+        Pair pair = new Pair();
+        return Flux.just(true)
+            .repeat()
+            .map(ignore -> {
+                byte[] buffer = new byte[BYTE_BUFFER_CHUNK_SIZE];
+                try {
+                    int numBytes = inputStream.read(buffer);
+                    if (numBytes > 0) {
+                        return pair.buffer(ByteBuffer.wrap(buffer, 0, numBytes)).readBytes(numBytes);
+                    } else {
+                        return pair.buffer(null).readBytes(numBytes);
+                    }
+                } catch (IOException ioe) {
+                    throw LOGGER.logExceptionAsError(new RuntimeException(ioe));
+                }
+            })
+            .takeUntil(p -> p.readBytes() == -1)
+            .filter(p -> p.readBytes() > 0)
+            .map(Pair::buffer);
     }
 
-    private static ByteBuffer toByteArray(InputStream in) {
-        try {
-            ByteArrayOutputStream os = new ByteArrayOutputStream();
-            byte[] buffer = new byte[1024];
-            int len;
-
-            // read bytes from the input stream and store them in buffer
-            while ((len = in.read(buffer)) != -1) {
-                // write bytes from the buffer into output stream
-                os.write(buffer, 0, len);
-            }
-            return ByteBuffer.wrap(os.toByteArray());
-        } catch (IOException e) {
-            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+    private static class Pair {
+        private ByteBuffer byteBuffer;
+        private int readBytes;
+
+        ByteBuffer buffer() {
+            return this.byteBuffer;
+        }
+
+        int readBytes() {
+            return this.readBytes;
+        }
+
+        Pair buffer(ByteBuffer byteBuffer) {
+            this.byteBuffer = byteBuffer;
+            return this;
+        }
+
+        Pair readBytes(int cnt) {
+            this.readBytes = cnt;
+            return this;
         }
     }
+
+    /**
+     * Compare if a byte value equals to a hex type value.
+     *
+     * @param byteValue the byte type value
+     * @param hexValue the hex type value
+     * @return true if two type's values are equal in unsigned int comparision.
+     */
+    private static boolean isEqual(byte byteValue, int hexValue) {
+        return Byte.toUnsignedInt(byteValue) == Byte.toUnsignedInt((byte) hexValue);
```

Comment:
If we do want to end up keeping this we shouldn't cast and re-cast the hex value.

`byteValue == (byte) hexValue` should be sufficient.